### ISSUE TITLE
ClassCastException: com.sun.javafx.geom.BoxBounds -> com.sun.javafx.geom.RectBounds

### ISSUE THREAD

**Issue by jovial** [scalafx/scalafx#265]
I get:

`java.lang.ClassCastException: com.sun.javafx.geom.BoxBounds com.sun.javafx.geom.BoxBounds cannot be cast to com.sun.javafx.geom.RectBounds`

when running https://github.com/scalafx/ProScalaFX/blob/master/src/proscalafx/ch04/reversi/ui/Reversi.scala

This occurs when moving the mouse over the board.

the stacktrace looks like:

```
java.lang.ClassCastException: com.sun.javafx.geom.BoxBounds cannot be cast to com.sun.javafx.geom.RectBounds
	at com.sun.javafx.sg.prism.NGNode.applyTransform(NGNode.java:2465)
	at com.sun.javafx.sg.prism.NGNode.accumulateGroupDirtyRegion(NGNode.java:1207)
	at com.sun.javafx.sg.prism.NGNode.accumulateDirtyRegions(NGNode.java:1022)
	at com.sun.javafx.sg.prism.NGNode.accumulateGroupDirtyRegion(NGNode.java:1189)
	at com.sun.javafx.sg.prism.NGNode.accumulateDirtyRegions(NGNode.java:1022)
	at com.sun.javafx.sg.prism.NGNode.accumulateGroupDirtyRegion(NGNode.java:1189)
	at com.sun.javafx.sg.prism.NGNode.accumulateDirtyRegions(NGNode.java:1022)
	at com.sun.javafx.sg.prism.NGNode.accumulateGroupDirtyRegion(NGNode.java:1189)
	at com.sun.javafx.sg.prism.NGNode.accumulateDirtyRegions(NGNode.java:1022)
	at com.sun.javafx.sg.prism.NGNode.accumulateGroupDirtyRegion(NGNode.java:1189)
	at com.sun.javafx.sg.prism.NGNode.accumulateDirtyRegions(NGNode.java:1022)
	at com.sun.javafx.sg.prism.NGNode.accumulateGroupDirtyRegion(NGNode.java:1189)
	at com.sun.javafx.sg.prism.NGNode.accumulateDirtyRegions(NGNode.java:1022)
	at com.sun.javafx.tk.quantum.ViewPainter.paintImpl(ViewPainter.java:246)
	at com.sun.javafx.tk.quantum.PresentingPainter.run(PresentingPainter.java:91)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at com.sun.javafx.tk.RenderJob.run(RenderJob.java:58)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at com.sun.javafx.tk.quantum.QuantumRenderer$PipelineRunnable.run(QuantumRenderer.java:125)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by jovial** [comment]
This appears to be caused by the 3d transforms used on the board `GridPane`

**Comment by jpsacha** [comment]
The same exception is shown in the original JavaFX code from the first edition of the book, "Pro JavaFX 2", when run on Java 8. The exception is only a warning, example works fine. The original code was created for JavaFX 2. When run with JavaFX 2 there was no exception. The issue is with the code that uses JavaFX 3D to show the board under a projective transformation. 

Interestingly, in the "Pro JavaFX 8", the code related to 3D was removed, possibly due to the above exception.  I pushed update to ProScalaFX repo that comments out that code to avoid runtime warnings.
